lib/Widgets/Settings.vala

```diff
@@ -69,6 +73,36 @@ namespace Granite {
             }
         }
 
+        /**
+         * Possible clock format preferences expressed by the user
+         */
+        public enum ClockFormat {
+            /**
+             * The user prefers a 12 hour clock
+             */
+            12H,
+            /**
+             * The user prefers a 24 hour clock
+             */
+            24H
+        }
+
+        private ClockFormat? _clock_format = null;
+        /**
+         * Whether the user would prefer a 12 hour or 24 hour clock
+         */
+        public ClockFormat clock_format {
+            get {
+                if (_clock_format == null) {
+                    setup_clock_format ();
+                }
+                return _clock_format;
```

Comment:
```suggestion
                return _clock_format;
```
_clock_format might be null, so we need to explicitely do/
`return _clock_format ?? ClockFormat.24H`